Integrate Slint Tree Sitter code

jrmoulton opened the pull request:
This integrates the customizable parts of the Tree Sitter parser for Slint into the main Slint repo

hunger commented on editor/README.md:
```diff
@@ -121,17 +121,16 @@ exists under the `language` server section:
-You might also find the `jrmoulton/tree-sitter-slint` tree-sitter parser for the Slint language
-interesting. Unfortunately this is not integrated into the tree-sitter support of Neovim at this
-time.
+You can also install the Tree Sitter parser for Slint using `TSInstall slint`
```
... if you have nvim-treesitter installed.

hunger replied:
Will that still work after moving the code over into the slint repository?

hunger commented on editor/neovim/tree-sitter-slint/binding.gyp:
```diff
@@ -0,0 +1,19 @@
+{
```
Should we have (some|all) of these files in editors/tree-sitter? The parser itself can be used for more than neovim, can't it?

hunger replied:
Can you add a license header into the file?

jrmoulton replied:
> Should we have (some|all) of these files in editors/tree-sitter? The parser itself can be used for more than neovim, can't it?

Yes. So that would make more sense


> Can you add a license header into the file?

Yep. I'll get that done.

hunger commented:
Is there anything I can do to help with this at this time?

jrmoulton commented:
Sorry I haven't responded. School has gotten super busy lately and I haven't looked at this PR in a bit. I'll probably have some time on Sunday to get those changes done.

> I guess you can not add a comment into the scm files, can you? All files that can have a copyright header should have one!

Comments can be added. They are prefixed with a `;`
I'll get the license added.

jrmoulton commented:
I've added the licenses and moved the folder. 
I left the folder name tree-sitter-slint because that is the tree sitter recommended format... but it might not matter here at all and it seems slightly redundant to me. Thoughts on renaming it to editor/tree-sitter?

hunger commented:
Yes, it is probably a bit redundant, but let's just follow the tree-sitter conventions.

Looks good to me! I need to update the lecense xtask now that the files are in the expected place.

hunger commented:
I pushed a patch to make our CI copyright lawyers happy to here:

https://github.com/jrmoulton/slint/pull/1
